src/System.Text.Json/src/System/Text/Json/Serialization/JsonSerializer.Read.cs

```diff
@@ -75,7 +74,17 @@ public static partial class JsonSerializer
                 }
                 else if (tokenType == JsonTokenType.StartObject)
                 {
-                    HandleStartObject(options, ref reader, ref state);
+                    if (!state.Current.IsProcessingProperty)
```

Comment:
Seems like this would also have to be done for EndObject\EndArray

Reply:
The JsonDocument eats it.